lib/array_intersection.rb

```diff
@@ -1,3 +1,25 @@
 def intersection(list1, list2)
-  raise NotImplementedError, "Intersection not implemented"
+  lookup_hash = {}
+  
+  if list1.length < list2.length
+    list1.each do |num|
+      lookup_hash[num] = true
+    end
+
+    intersection_array = list2.map do |num|
+      num if lookup_hash[num]
+    end
+
+    return intersection_array.compact
```

Comment:
Use `map` if you want to do an operation on each element in the array.  In this case you want to get only **some** of the elements of the array.

```suggestion
    intersection_array = list2.select do |num|
       lookup_hash[num]
    end

    return intersection_array
```